CRM/Case/BAO/Case.php

```diff
@@ -529,6 +529,9 @@ public static function retrieveCaseIdsByContactId($contactID, $includeDeleted =
     if (!$includeDeleted) {
       $query .= " AND ca.is_deleted = 0";
     }
+    if (isset($statusId)) {
+      $query .= " AND ca.status_id = '$statusId'";
+    }
```

Comment:
To expand on that - the reason it needs to be passed as a variable in an array is that way it will be escaped in a way that will prevent any security vulnerabilities - e.g adding a subquery into the status id. Probably this field has already been validated but we should rely on that.